TempFileGZIPWriter.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace samdark\sitemap;
+
+/**
+ * Flushes buffer into temporary stream and compresses stream into a file on finish
+ */
+class TempFileGZIPWriter
+    implements WriterInterface
+{
+    /**
+     * @var string Name of target file
+     */
+    private $filename;
+
+    /**
+     * @var ressource for php://temp stream
+     */
+    private $tempFile;
+
+    public function __construct($filename)
+    {
+        $this->filename = $filename;
+        $this->tempFile = fopen('php://temp/', 'w');
+    }
+
+    /**
+     * Store data in a temporary stream/file
+     *
+     * @param string $data
+     */
+    public function append($data)
+    {
+        assert(isset($this->tempFile));
+
+        fwrite($this->tempFile, $data);
+    }
+
+    /**
+     * Deflate buffered data
+     */
+    public function finish()
+    {
+        $file = fopen('compress.zlib://' . $this->filename, 'w');
+        rewind($this->tempFile);
+        stream_copy_to_stream($this->tempFile, $file);
+
+        fclose($file);
+        fclose($this->tempFile);
+        unset($this->tempFile);
```

Comment:
Was `unlink` meant here?

Reply:
No, `unset` is correct. Just like I do `unset($this->file)` in the other writer back ends.
I would guess that unlinking `'php://temp/'` is not necessary anyway, as it was not done in the original code either.